packages/lmi/src/lmi/types.py

```diff
@@ -72,10 +71,33 @@ class LLMResult(BaseModel):
     messages: list[Message] | None = Field(
         default=None, description="Messages received from the LLM."
     )
-    prompt_count: int = 0
-    completion_count: int = 0
+    prompt_count: int | None = None
+    completion_count: int | None = None
     model: str
     date: str = Field(default_factory=datetime.now().isoformat)
+
+    # Cached token counts - extracted from provider-specific usage fields:
+    # - Both providers report cache reads via prompt_tokens_details.cached_tokens
+    # - Only Anthropic reports cache creation via cache_creation_input_tokens
+    cache_read_tokens: int | None = Field(
```

Comment:
Let's make all the `_tokens` statements frozen to avoid statefulness issues (where `cost` gets de-sync'd from token counts)